src/core/qgsproperty.h

```diff
@@ -125,25 +127,57 @@ class CORE_EXPORT QgsPropertyDefinition
      * \param description can be any localised string describing what the property is used for.
      * \param helpText parameter should specify a descriptive string for users outlining the types
      * of value acceptable by the property (eg 'dashed' or 'solid' for a line style property).
+     * \param origin The origin of the property
+     * \param comment A free comment for the property
      */
-    QgsPropertyDefinition( const QString &name, DataType dataType, const QString &description, const QString &helpText );
+    QgsPropertyDefinition( const QString &name, DataType dataType, const QString &description, const QString &helpText, const QString &origin = QString(), const QString &comment = QString() );
 
     /**
      * Returns the name of the property. This is used internally and should be a unique, alphanumeric string.
      */
     QString name() const { return mName; }
 
     /**
+     * Sets the name of the property
+     */
+    void setName( const QString &name ) { mName = name; }
+
+    /**
+     * Returns the origin of the property
```

Comment:
Would be great if you could add a tiny bit more information here (explanation or examples)

Reply:
Indeed. Done.